### PR TITLE
[ISSUE #2660]  Add the type parameter when call publishConfig(), and support Aliyun ACM

### LINKED ISSUE #2660: NacosConfigService#publishConfig缺失配置类型字段
**Issue Description**
环境
Nacos1.2.1

**Describe what happened (or what feature you want)**
问题描述
com.alibaba.nacos.client.config.NacosConfigService#publishConfig 默认没有为配置内容设置type, 导致在dashboard中默认显示text类型

**Describe what you expected to happen**
增加type类型, 使dashboard能正确显示

### PR DESCRIPTION
**Pull request by flyer5200**
## What is the purpose of the change

Fixes #2660 

## Brief changelog

See ISSUE #2660 

1. Add parammeter check and support Aliyun ACM
2. Add the type parameter when the [com.alibaba.nacos.client.config.NacosConfigService#publishConfig](https://github.com/alibaba/nacos/blob/4e52ff050abb65a7cb14a09a2d32641ef3505aa0/client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java#L110) method is executed so that the naco or Aliyun ACM dashboard can be displayed correctly

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java [hunk 1]
```diff
@@ -68,10 +69,12 @@
     private ClientWorker worker;
     private String namespace;
     private String encode;
+    private String configType;
     private ConfigFilterChainManager configFilterChainManager = new ConfigFilterChainManager();
 
     public NacosConfigService(Properties properties) throws NacosException {
         ValidatorUtils.checkInitParam(properties);
+        configType = properties.getProperty(PropertyKeyConst.CONFIG_DATA_TYPE, ConfigType.JSON.getType());
```
Comment (chuntaojun) [inline client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java:77]: 并不需要作为一个成员属性来保存，方法传一个默认值就好

File: client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java [hunk 2]
```diff
@@ -230,6 +233,10 @@ private boolean publishConfigInner(String tenant, String dataId, String group, S
         params.add(group);
         params.add("content");
         params.add(content);
+        //if type is empty, set a default value: json
```
Comment (chuntaojun) [inline client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java:236]: Why is the json type the default value, I think text the default value

Comment (flyer5200) [reply client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java:236]: text is also ok, I think json format is more readable in most scenarios

File: client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java (hunk 1 above)
Comment (flyer5200) [reply client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java:77]: Use the default value? How does the client pass in a custom type? Need to modify the interface definition to add an overload method?

Comment (chuntaojun) [reply client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java:77]: Yes, you just need to add a publishConfig method

File: client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java (hunk 2 above)
Comment (chuntaojun) [reply client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java:236]: I want to be able to follow the original design, the client and the service side of the unity

Comment (flyer5200) [reply client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java:236]: Okay, I will add another commit

**Comment by chuntaojun** [comment]
A PR can only solve one issue, and ACM can solve another one

**Comment by KomachiSion** [comment]
The PR has no update for a long time, Any process? If no more response, I will close it.

**Comment by horizonzy** [comment]
in server, if request not set type, maybe it can set default type value 'text'. it will cover old java client and other request without set type's value

**Pull request closed without merging**